src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -1275,25 +1276,40 @@ private static boolean isAliasValid(String alias) {
         return invalid;
     }
 
+    private Iterator<Resource> queryAllVanityPaths(String query) {
+        log.debug("start vanityPath query: {}", query);
+        long queryStart = System.nanoTime();
+        final Iterator<Resource> i = resolver.findResources(query, "JCR-SQL2");
+        long queryElapsed = System.nanoTime() - queryStart;
```

Comment:
This will probably not do what you want, because JCR queries are processed lazily (while iterating, not when executed). So, I would not log at this stage. Instead, I would measure the time on a higher level, while iterating over the result.

Reply:
We do that as well (and have been doing so...)